src/auth/googleauth.ts

```diff
@@ -752,7 +754,14 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
   }
 
   private async getCredentialsAsync(): Promise<CredentialBody> {
-    await this.getClient();
+    const client = await this.getClient();
+
+    if (
+      client instanceof BaseExternalAccountClient &&
+      client.getServiceAccountEmail()
+    ) {
+      return {client_email: client.getServiceAccountEmail() as string};
```

Comment:
what if you check whether or not this is null in the `if`, then you don't need `as string`.